components/modal/modal-backdrop.component.ts

```diff
@@ -12,12 +12,11 @@ export class ModalBackdropOptions {
 
 @Component({
   selector: 'bs-modal-backdrop',
-  template: ''
+  template: '',
+  host: {'class': `${ClassName.BACKDROP}`}
```

Comment:
Because it breaks modals' backdrop. `@HostBinding` and this approach are not the same thing. This initializes `class` with static value, while `@HostBinding` sets up a data binding, which overrides `class` value on every change detection run.

PS The "not recommended" thing in Angular Style Guide is not 100% correct. While it's better to use decorators for property and event binding, it's still more convenient to use this approach for static attribute initialization.